### ISSUE TITLE
Subtitle renaming behavior in #5577 still present in v4

### ISSUE THREAD

**Issue by tehniemer** [Sonarr/Sonarr#5890]
### Is there an existing issue for this?

- [x] I have searched the existing open and closed issues

### Current Behavior

Subtitle tags are being stripped from existing subtitle files in the same manner as what is reported in #5577 which was supposed to be fixed with #5578

When renaming files, subtitle types like "forced", "sdh" and "hi" are stripped from the filename and replaced with numbers 1, 2, 3, etc.

### Expected Behavior

Renaming subtitle files preserves the subtitle types. 

**Example, original files:**
xyz.en.srt
xyz.en.forced.srt
xyz.en.sdh.srt
xyz.en.hi.srt

**should be renamed to:**
abc.en.srt
abc.en.forced.srt
abc.en.sdh.srt
abc.en.hi.srt

### Steps To Reproduce

1. Add a subtitle file with one of the affect tags to an existing series (ex: Loudermilk - S01E01 - A Girl in Trouble Is a Temporary Thing WEBDL-1080p.no.forced)
2. Refresh the series to make Sonarr pick up the new subtitle
3. Trigger a rename
4. See the tag disappear from the filename

### Environment

```markdown
- OS: Debian 12
- Sonarr: 4.0.0.602
- Docker Install: Yes
- Using Reverse Proxy: Traefik V2
- Browser:
```

### What branch are you running?

Develop

### Trace Logs?

[sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/12286380/sonarr.trace.txt)

**Comment by markus101** [comment]
I am unable to reproduce this, language tags (as they're called internally) are properly transferred to the new file name. Before renaming you should be able to see them in the DB and with the commit that closed this issue additional logging has been added. If you can provide trace logs that show Sonarr parsed the language tags, but didn't use them in the new file name we can take another look.

**Comment by tehniemer** [comment]
I see some files in the database with the proper language tags, these do get renamed correctly.
![image](https://github.com/Sonarr/Sonarr/assets/11021632/506ae06a-8680-4377-8194-f9afa055276a)

Here are logs from renaming the series with LanguageTags showing them being preserved
[sonarr.debug.txt](https://github.com/Sonarr/Sonarr/files/12313479/sonarr.debug.txt)
[sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/12313480/sonarr.trace.txt)
[sonarr.txt](https://github.com/Sonarr/Sonarr/files/12313481/sonarr.txt)

But some are `NULL` and these don't.
![image](https://github.com/Sonarr/Sonarr/assets/11021632/41d33224-15b0-4c8a-bdf1-45db19ac7b6b)

Here are logs from renaming the series with `NULL` LanguageTags showing the tags in the filenames being dropped.
[sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/12312574/sonarr.trace.txt)
[sonarr.debug.txt](https://github.com/Sonarr/Sonarr/files/12312575/sonarr.debug.txt)
[sonarr.txt](https://github.com/Sonarr/Sonarr/files/12312576/sonarr.txt)

One interesting observation is that I changed the language from `.eng.` to `.en` on one file before the `NULL` rename and that file was renamed properly by keeping the `.sdh` tag, all others that were left as `.eng.` had `.sdh` stripped.

**Comment by tehniemer** [comment]
@markus101 do you need anything else from me to re-open this and look into it?

**Comment by markus101** [comment]
Your logs and information show that Sonarr is behaving correctly, when the value is `NULL` the file was either added before Sonarr has support for Language tags or was added to an existing file which was fixed by #5578.

When you renamed the file from `.eng.` to `.en`  that got Sonarr to see it as a new file and parsed it as such, which it wouldn't have before #5578.

Sonarr did not go back and process the existing files during a migration because that can be slow which leads to other issues. We'll need to discuss whether a migration or other task makes sense in this case.

**Comment by tehniemer** [comment]
@markus101 that seems like a reasonable explanation, these files were added only a few weeks ago, but in v3.

Is there any way for me to manually trigger a migration? I'd like to get these tags added to the database so I can do a mass rename without losing information.
